src/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -1460,13 +1460,27 @@ public static BigInteger Pow(BigInteger value, int exponent)
             if (bits == null)
                 return sign;
 
-            ulong h = bits[bits.Length - 1];
-            ulong m = bits.Length > 1 ? bits[bits.Length - 2] : 0;
-            ulong l = bits.Length > 2 ? bits[bits.Length - 3] : 0;
+            int length = bits.Length;
+
+            // Maximum exponent for double is 1023, uint bit length 32 => all bigint with bits[] longer 32 is infinity
+            // cases when exponent is 1024..1035 handled in NumericsHelpers.GetDoubleFromParts
+            const int infinity_length = 1024/kcbitUint;
```

Comment:
@mellinoe I see my code one more time and noticed `infinity_lengt`. This is not code style complient. Should I rename it?